src/memory/primitives/relocatable.test.ts

```diff
@@ -0,0 +1,99 @@
+import { test, expect, describe } from 'bun:test';
+import { Felt } from './felt';
+import {
+  Relocatable,
+  OffsetUnderflow,
+  SegmentError,
+  ForbiddenOperation,
+} from './relocatable';
+import { UnsignedInteger } from './uint';
+
+describe('Relocatable', () => {
```

Comment:
For most of these tests we assert that the offset moved, should we also assert that the segment didn't move?

Reply:
mhm good question, ok why not